title_belt_nhl/tests/test_schedule.py

```diff
@@ -134,3 +159,34 @@ def test_find_nearest_path_games_big(self, monkeypatch, league_schedule_big):
         for i, m in enumerate(expected):
             assert str(path_matches[i]) == str(m)
             assert path_matches[i].date_obj == m.date_obj
+
+    def test_find_nearest_path_v2(self, monkeypatch, league_schedule_all):
+        m = Mock()
+        m.return_value = league_schedule_all
+        monkeypatch.setattr("title_belt_nhl.schedule.getFullSchedule", m)
+
+        monkeypatch.setattr("title_belt_nhl.schedule.INITIAL_BELT_HOLDER", "FLA")
+
+        schedule = Schedule("MIN", from_date=date(2024, 10, 3))
+        assert schedule.belt_holder == "FLA"
+
+        start_match = schedule.find_match(
+            schedule.belt_holder, schedule.from_date.serial_date
+        )
+
+        path_matches = schedule.find_nearest_path_v2([[start_match]])
+
+        for i, m in enumerate(path_matches):
+            print(f"{path_matches[i].date_obj} {path_matches[i]}")
+        assert len(path_matches) == 6
+
+        m1 = Match("FLA", "BOS", date_obj=date(2024, 10, 8))
+        m2 = Match("OTT", "FLA", date_obj=date(2024, 10, 10))
+        m3 = Match("BUF", "FLA", date_obj=date(2024, 10, 12))
+        m4 = Match("PIT", "BUF", date_obj=date(2024, 10, 16))
+        m5 = Match("CBJ", "BUF", date_obj=date(2024, 10, 17))
+        m6 = Match("CBJ", "MIN", date_obj=date(2024, 10, 19))
+        expected = [m1, m2, m3, m4, m5, m6]
+        for i, m in enumerate(expected):
+            assert str(path_matches[i]) == str(m)
```

Comment:
TODO: also check `belt_holder` property is as-expected for all of these tests.

Reply:
Added this test for `test_find_nearest_path_v2`.  Seems to be working after returning a deepcopy of the returned `Match` from `Schedule.find_match`.  [Link](https://github.com/kawa2287/title-belt-nhl/blob/15862411a6d87741a000c8b6912fc34c3cc837a8/title_belt_nhl/schedule.py#L142C17-L142C39)